### PR TITLE
Add decimal support and test cases for arithmetic decimal function "power"

### PR DESCRIPTION
**Pull request by anshuldata**
* Enhance framework to honor type with arguments (e.g. decimal<38, 1>) and add support for decimal type
* Added test cases for power function having decimal arguments in duckdb and snowflake dialect

### CODE REVIEW

**Comment by anshuldata** [comment]
@richtia kindly review

File: cases/arithmetic/power_decimal.yaml
Comment (richtia) [inline cases/arithmetic/power_decimal.yaml:1]: Let's create a new folder called arithmetic_decimal.yaml to store these new function variations that take in decimal arg types to better align with substrait.

File: cases/arithmetic/power_decimal.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  - group: complex_number
+    args:
+      - value: -1
+        type: decimal
+      - value: 0.5
+        type: decimal<38,1>
+    options:
+      complex_number_result: ERROR
+    result:
+      special: error
```
Comment (richtia) [inline cases/arithmetic/power_decimal.yaml:94]: newline

Comment (anshuldata) [reply cases/arithmetic/power_decimal.yaml]: fixed it

File: cases/arithmetic/power_decimal.yaml
Comment (anshuldata) [reply cases/arithmetic/power_decimal.yaml:1]: fixed it

File: bft/dialects/types.py [hunk 2]
```diff
@@ -136,16 +136,18 @@ def __supports_case_kernel(
...
-                        if arg.type not in self.supported_types:
+                        if type_to_check not in self.supported_types:
                             matched = False
                             break
-                        any_map[ktype] = arg.type
-                    elif any_map[ktype] != arg.type:
+                        any_map[ktype] = type_to_check
+                    elif any_map[ktype] != type_to_check:
                         matched = False
                         break
```
Comment (richtia) [inline bft/dialects/types.py:152]: There have been a lot of changes in this block over the last few PRs.  Could you add some comments for this.

File: bft/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+def type_to_dialect_type(type: str, type_map: Dict[str, str]):
+    type_to_check = type.split("<")[0].strip() if "<" in type else type
+    if type_to_check not in type_map:
+        raise Exception(f"Unrecognized type: {type}")
+    type_val = type_map[type_to_check]
+    if not "<" in type:
+        return type_val
+    # transform parameterized type name to have dialect type
+    return type.replace(type_to_check, type_val).replace("<", "(").replace(">", ")")
```
Comment (richtia) [inline bft/utils/utils.py:12]: Could you add docstings for this function and as well as typehints for the return.

File: bft/dialects/types.py (hunk 2 above)
Comment (anshuldata) [reply bft/dialects/types.py]: added comment

File: bft/utils/utils.py (hunk 3 above)
Comment (anshuldata) [reply bft/utils/utils.py]: added comment

**Review by richtia — APPROVED** [review]

**Comment by anshuldata** [comment]
@richtia can your merge this please. I don't have wire access. Thanks

**Comment by anshuldata** [comment]
Thanks @richtia

**Pull request merged**